pkg/storage/pebble_mvcc_scanner.go

```diff
@@ -479,6 +492,11 @@ func (p *pebbleMVCCScanner) getAndAdvance() bool {
 		// Note that this will trigger an error higher up the stack. We
 		// continue scanning so that we can return all of the intents
 		// in the scan range.
+		if p.err = p.memMonitor.Grow(
```

Comment:
This looked wrong to me until I realized that `p.intents` is a `pebble.Batch` and not a `pebbleResults`. Might be worth pointing out in a comment, I liked that we didn't have calls to `Grow` all over the place and I assume this is the only place where we write to `p.intents` so it's fine.